server/dbClient.js

```diff
@@ -3281,8 +3282,13 @@ const _updateAddedAssignments = async (db, username, addedAssignments, term, sem
         let weights = user.weights[term][semester][index].weights;
         let addedWeights = user.addedWeights[term][semester][index].weights;
         let actualAddedWeights = new Set();
+        // Iterate each assignment in the school class
         for (let j = 0; j < addedAssignments[i].data.length; j++) {
             let assignment = addedAssignments[i].data[j];
+            // Discard HTML tags
+            assignment.category = assignment.category.replace(/(<([^>]+)>)/gi, "");
+            addedAssignments[i].data[j].assignment_name = assignment.assignment_name.replace(/(<([^>]+)>)/gi, "");
```

Comment:
Yes I believe modifying `assignment.assignment_name` modifies a copy of the data rather than the data that is sent to the database, whereas category is handled differently so it works

Reply:
While the category is additionally written to a different place, it is still part of the assignment and should update in the same way as the name. There should be a consistent way to modify the two fields.